test/testlib.py

```diff
@@ -414,16 +427,56 @@ def tearDown(self):
         elif self.machine.address:
             self.check_journal_messages()
 
-    def start_cockpit(self):
+    def wait_for_cockpit_running(self, atomic_wait_for_host="localhost"):
+        """Wait until cockpit is running.
+
+        We only need to do this on atomic systems.
+        On other systems, systemctl blocks until the service is actually running.
+        """
+        if not "atomic" in self.machine.os or not atomic_wait_for_host:
+            return
+        WAIT_COCKPIT_RUNNING = """#!/bin/sh
+until curl -s --connect-timeout 1 http://%s:9090 >/dev/null; do
+    sleep 0.5;
+done;
+""" % (atomic_wait_for_host)
+        with Timeout(seconds=30, error_message="Timeout while waiting for cockpit/ws to start"):
+            self.machine.execute(script=WAIT_COCKPIT_RUNNING)
+
+    def start_cockpit(self, atomic_wait_for_host="localhost"):
         """Start Cockpit.
 
         Cockpit is not running when the test virtual machine starts up, to
         allow you to make modifications before it starts.
         """
-        self.machine.execute("systemctl start cockpit-testing.socket")
+        if "atomic" in self.machine.os:
+            # HACK
```

Comment:
@dperpeet Have you filed a bug about this upstream? If so, can you include the link here?